File: 0009-palindrome-number/0009-palindrome-number.ts
function isPalindrome(x: number): boolean {
    if (x < 0) return false;

    let num = x;

    let rev = 0;
    while(num > 0) {
        let ld = num % 10;
        rev = ld + (rev * 10);
        num = Math.floor(num / 10);
    }

    return rev === x;
};

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x < 0) return false;

        int n = x;
        int rev = 0;
        while(n > 0) {
            int ld = n % 10;
            rev = ld + (rev * 10);
            n = n / 10;
        }

        return x == rev;
    }
}